x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/AbstractUpgradeTestCase.java

```diff
@@ -82,22 +86,30 @@ protected Settings restClientSettings() {
 
     @Before
     public void setupForTests() throws Exception {
-        awaitBusy(() -> {
-            boolean success = true;
+        assertBusy(() -> {
+            final List<String> missingTemplates = new ArrayList<>();
+
             for (String template : templatesToWaitFor()) {
                 try {
                     final Request headRequest = new Request("HEAD", "_template/" + template);
                     headRequest.setOptions(allowTypesRemovalWarnings());
+
                     final boolean exists = adminClient()
                         .performRequest(headRequest)
-                            .getStatusLine().getStatusCode() == 200;
-                    success &= exists;
+                        .getStatusLine().getStatusCode() == 200;
+
+                    if (!exists) {
```

Comment:
nit: `== false` please

Reply:
Roger. That'll take a little getting used to :-)